controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -37,14 +38,15 @@ func (r *AKODeploymentConfigReconciler) initAVI(
 	// Lazily initialize aviClient so we don't skip other reconciliations
 	if r.aviClient == nil {
```

Comment:
You can change this to also check if the certificate or credentials changed or not to re-init your avi client